backend/lm_backend/api/booking.py

```diff
@@ -96,9 +105,30 @@ async def get_bookings_job(job_id: str):
     return [BookingRow.parse_obj(x) for x in fetched]
 
 
-async def _is_booking_available(booking: Booking):
-    """Check if the total available is greater than the sum of in_use_total and new_booked, if it is then
-    there is no more bookings available. The in_use_total is calculated accounting for all bookings.
+async def _get_limit_for_booking_feature(product_feature: str) -> int:
+    """
+    Get the maximum amount of licenses that can be booked.
+    If the limit is the same as the total amount of licenses, all licenses can be booked.
+    """
+    product, feature = product_feature.split(".")
+
+    # Use product name to fetch config item from database
+    query = config_table.select().where(config_table.c.product == product)
+    fetched = await database.fetch_one(query)
+    config_row = ConfigurationRow.parse_obj(fetched)
+    config_item = ConfigurationItem(
+        **config_row.dict(exclude={"features"}), features=literal_eval(config_row.features)
```

Comment:
Thinking more about this and looking at the code, I think it would be very helpful to constraint the serialization/deserialization of the features to only insertion and extraction from the database. You could set it up so that throughout the rest of the code, you are dealing with a nested pydantic object, and it's only when you actually go to save it into the database that it's converted to a string.

On that note, we also really need to figure out how to get JSON and JSONB working in our apps. It's a perfect fit for this use-case.

Reply:
I didn't. I followed this pattern of excluding the features string and evaluating it to a dict because it's the way it's being handled in all the other parts of the code.
But again, we should take some time to look into refactoring it.